docs/en/tutorial.md

```diff
@@ -341,7 +341,169 @@ Let's run some queries to analyze the 2M rows of data...
     │ 2015-07-01 00:41:48 │ 2015-07-01 00:44:45 │          6.3 │                 -94 │                  132 │ JFK          │ 2015 │   1 │    0 │
     │ 2015-07-01 01:06:18 │ 2015-07-01 01:14:43 │        11.76 │                  37 │                  132 │ JFK          │ 2015 │   1 │    1 │
     ```
+## 4. Create a Dictionary
 
+If you are new to ClickHouse, it is important to understand how ***dictionaries*** work. A dictionary is a mapping of key->value pairs that is stored in memory.
+
+1. Let's see how to create a dictionary associated with a table in your ClickHouse service. The table, and therefore the dictionary, will be based on a CSV file that contains 265 rows, one row for each neighborhood in NYC. The neighborhoods are mapped to the names of the NYC boroughs (NYC has 5 boroughs: the Bronx, Brooklyn, Manhattan, Queens and Staten Island), and this file counts Newark Airport (EWR) as a borough as well.
+
+  This is part of the CSV file (shown as a table for clarity).  The `LocationID` column in the file maps to the `pickup_nyct2010_gid` and `dropoff_nyct2010_gid` columns in your `trips` table:
+
+    | LocationID      | Borough |  Zone      | service_zone |
+    | ----------- | ----------- |   ----------- | ----------- |
+    | 1      | EWR       |  Newark Airport   | EWR        |
+    | 2    |   Queens     |   Jamaica Bay   |      Boro Zone   |
+    | 3   |   Bronx     |  Allerton/Pelham Gardens    |    Boro Zone     |
+    | 4     |    Manhattan    |    Alphabet City  |     Yellow Zone    |
+    | 5     |  Staten Island      |   Arden Heights   |    Boro Zone     |
+
+
+2. The URL for the file is `https://datasets-documentation.s3.eu-west-3.amazonaws.com/nyc-taxi/taxi_zone_lookup.csv`. Run the following SQL, which creates a new table named `taxi_zone_source` and then populates the table from the CSV file in S3:
+  ```sql
+  CREATE TABLE taxi_zone_source (
+      LocationID UInt16 DEFAULT 0,
+      Borough String,
+      Zone String,
+      service_zone String
+    )
+    ENGINE = MergeTree
+    PRIMARY KEY LocationID
+  ```
+
+  ```sql
+  INSERT INTO taxi_zone_source
+  SELECT * FROM s3('https://datasets-documentation.s3.eu-west-3.amazonaws.com/nyc-taxi/taxi_zone_lookup.csv', 'CSVWithNames')
+  ```
+3. Create a dictionary associated with the table `taxi_zone_source`.
+
+  ```sql
+  CREATE DICTIONARY taxi_zone_dictionary
+  (
+    LocationID UInt16,
+    Borough String,
+    Zone String,
+    service_zone String
+  )
+  PRIMARY KEY LocationID
+  SOURCE(CLICKHOUSE(TABLE 'taxi_zone_source'))
```

Comment:
Maybe also to cover that only user `default` can create a dictionary without specifying a password? If a `default` user will create another user then it is required to specify it like this `USER another_user PASSWORD plain_text`